strongbox-distribution/src/main/resources/deb/control

```diff
@@ -0,0 +1,9 @@
+Package: strongbox
+Version: [[version]]
+Architecture: all
+Section: Miscellaneous
+Priority: low
+Homepage: https://github.com/strongbox/strongbox
+Description: Modern artifact repository manager
+Maintainer: Carlspring <carlspring@gmail.com>
+Depends: java8-runtime | java8-runtime-headless
```

Comment:
I think we probably shouldn't enforce these dependencies but rather have them as [`Suggests`](https://linux.die.net/man/5/deb-control) so people could opt-in to install them via `apt install --install-suggests`. 
There are valid cases where people already have a version of `JRE` installed on the server and enforcing the dependency could lead to unwanted upgrades and issues.

// @carlspring thoughts?

Reply:
Tough one... On the one hand -- you need Java installed, on the other -- you don't want to force a specific version of the JDK, as long as it's a recent 1.8 one.

Let's go for the `Suggests`.

Reply:
The JDK deb packages I looked at all took a dependency on the JRE, and the JREs all had `Provides` on the java8-runtime, headless or both, so I am pretty sure 

> There are valid cases where people already have a version of JRE installed on the server and enforcing the dependency could lead to unwanted upgrades and issues.

Can you describe in more detail what you mean here?

> you don't want to force a specific version of the JDK, as long as it's a recent 1.8 one.

I don't think this does, as long as the Provides field is filled in, which it should be.

Reply:
> Can you describe in more detail what you mean here?

By using `Depends` you are making `apt` require the `java8-runtime` package to be installed or otherwise Strongbox won't be installed. However, there are many ways to setup a server. You could extract a `JDK/JRE` in a specific directory (i.e. you've downloaded from `openjdk` or `oraclejdk`) and exported `JAVA_HOME` -- or you could have installed it via `apt install`. If you already have a custom JDK/JRE setup - then installing `java-runtime8` would make no sense. 

After having another look at the official docs, [`Recommends`](https://www.debian.org/doc/debian-policy/ch-relationships.html#binary-dependencies-depends-recommends-suggests-enhances-pre-depends) is a better choice (rather than my initial suggestion to use `Suggests`). This will install the the dependency by default when you do `apt install ./strongbox-distribution-1.0-SNAPSHOT.deb` but would allow the flexibility to add skip the jre dependency with when doing `apt --no-install-recommends install ./strongbox-distribution-1.0-SNAPSHOT.deb` (should it be necessary)

Reply:
Right, I had not considered about non-deb java installs.